bin/localeapp

```diff
@@ -18,150 +21,170 @@ if ENV['FAKE_WEB_DURING_CUCUMBER_RUN']
-  elsif File.exist?('.env') && IO.read('.env') =~ /^LOCALEAPP_API_KEY=(\w+)$/
-    $1
+module LocaleappGLIWrapper
+  gli2 = GLI::VERSION >= '2.0.0'
+  if gli2
+    include GLI::App
   else
-    nil
+    include GLI
   end
+  extend self
+
+  pre do |global_options, command, options, args|
+    global_options[:k] = if global_options[:k]
+      global_options[:k]
+    elsif ENV['LOCALEAPP_API_KEY']
+      ENV['LOCALEAPP_API_KEY']
+    elsif File.exist?('.env') && IO.read('.env') =~ /^LOCALEAPP_API_KEY=(\w+)$/
+      $1
+    else
+      nil
+    end
 
-  if Localeapp.has_config_file? || !global_options[:k].nil?
-    true
-  else
-    puts "Could not load config file and no key specified"
-    exit 1
+    if Localeapp.has_config_file? || !global_options[:k].nil?
+      true
+    else
+      puts "Could not load config file and no key specified"
+      exit 1
+    end
   end
-end
 
-desc "API Key (for when there is no configuration file)"
-flag [:k, 'api-key']
-
-version Localeapp::VERSION
-
-desc "Creates new configuration files and confirms key works"
-skips_pre
-arg_name "<api_key>"
-command :install do |c|
-  c.desc "install configuration files in .localeapp/"
-  c.switch [:s, 'standalone']
-
-  c.desc "create configuration when using localeapp via a heroku addon (PRE ALPHA)"
-  c.switch [:h, 'heroku']
-
-  c.desc "install a skeleton project suitable for Github (warning: README.md will be overwritten)"
-  c.switch [:g, 'github']
-
-  c.action do |global_options, options, args|
-    key = args.first
-    installer = Localeapp::CLI::Install.new
-    installer.config_type = :standalone if options[:standalone]
-    installer.config_type = :heroku if options[:heroku]
-    installer.config_type = :github if options[:github]
-    installer.config_type ||= :rails
-    unless installer.execute(key)
-      exit_now! "", 1
+  desc "API Key (for when there is no configuration file)"
+  flag [:k, 'api-key']
+
+  version Localeapp::VERSION
+
+  desc "Creates new configuration files and confirms key works"
+  skips_pre
+  arg_name "<api_key>"
+  command :install do |c|
+    # FIXME:
+    install_opts = {}
+
+    c.desc "install configuration files in .localeapp/"
+    c.switch [:s, 'standalone']
+
+    c.desc "create configuration when using localeapp via a heroku addon (PRE ALPHA)"
+    c.switch [:h, 'heroku']
+
+    c.desc "install a skeleton project suitable for Github (warning: README.md will be overwritten)"
+    c.switch [:g, 'github']
+
+    # FIXME:
+    c.desc "write API key to file at PATH"
+    c.arg_name "PATH"
+    c.default_value Localeapp::ENV_FILE_PATH
+    c.flag([:e, "write-env-file"]).tap do |opt|
+      opt.define_singleton_method :argument_name do
+        "[PATH]"
+      end
+      opt.define_singleton_method :arguments_for_option_parser do
+        ["-e", "--write-env-file [PATH]"]
+      end
+      opt.define_singleton_method :handler do |path|
+        install_opts[:write_env_file] = path || Localeapp::ENV_FILE_PATH
+      end
+    end
```

Comment:
`gli` uses `optparse`, but many basic features seems broken, or I
didn't found how to do it with `gli`.

For examples here we want the equivalent of `"-e", "--write-env-file
[PATH]"`, which does not seem possible only with the `#flag` DSL
method. No class in `gli` seems to implement on option with optional
argument, I tried to use `GLI::CommandLineOption`, but it's not easy
because gli has code like `kind_of? Flag` in a few places instead of
duck typing (in that case , using `GLI::CommandLineOption` get the
behavior we want, but breaks the formatting for option help in the
usage). Ugly hack:
``` ruby
c.desc "write API key to file at PATH"
c.arg_name "PATH", :optional
c.default_value Localeapp::ENV_FILE_PATH

env_option = GLI::CommandLineOption.new(
  [:e, "env-file"],
  desc:           "write API key to file at PATH (default: `.env')",
  long_desc:      "foo",
  default_value:  ".env",
  arg_name:       "PATH"
)
env_option.define_singleton_method :arguments_for_option_parser do
  ["-e", "--env-file [PATH]"]
end
env_option.define_singleton_method :required? do
  false
end
env_option.define_singleton_method :argument_name do
  "[PATH]"
end
env_option.associated_command = c
c.flags[env_option.name] = env_option
c.clear_nexts
c.flags_declaration_order << env_option
```

I also tried to use `optparse` directly, since `gli` rely on it:
``` ruby
c.desc "write API key to file at PATH"
c.arg_name "PATH"
c.default_value Localeapp::ENV_FILE_PATH
c.flag([:e, "write-env-file"]).tap do |opt|
  opt.define_singleton_method :argument_name do
    "[PATH]"
  end
  opt.define_singleton_method :arguments_for_option_parser do
    ["-e [PATH]", "--write-env-file [PATH]"]
  end
  opt.define_singleton_method :handler do |path|
    install_opts[:write_env_file] = path || Localeapp::ENV_FILE_PATH
  end
end

OptionParser.new do |opts|
  opts.on "-e", "--write-env-file [PATH]" do |e|
    install_opts[:write_env_file] = e || Localeapp::ENV_FILE_PATH
  end
end.parse!
```
But now the problem is `ARGV` mutations, `gli` does not expect it to
happen, and if we use `optparse` without modifying `ARGV`, the
problematic option will still cause troubles with `gli`.


In the end, we just need to be able to give our option handler to
`gli`, but I didn't found how.